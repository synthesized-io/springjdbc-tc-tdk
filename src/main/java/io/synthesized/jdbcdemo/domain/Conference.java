package io.synthesized.jdbcdemo.domain;

import lombok.Data;

@Data
public class Conference {
    private final int id;
    private final String name;
}
