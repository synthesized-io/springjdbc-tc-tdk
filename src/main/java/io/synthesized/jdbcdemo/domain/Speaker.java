package io.synthesized.jdbcdemo.domain;

import lombok.Data;

@Data
public class Speaker {
    private final int id;
    private final String name;
}
