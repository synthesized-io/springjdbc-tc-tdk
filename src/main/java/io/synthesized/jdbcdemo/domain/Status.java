package io.synthesized.jdbcdemo.domain;

public enum Status {
    NEW,
    IN_REVIEW,
    ACCEPTED,
    REJECTED
}
