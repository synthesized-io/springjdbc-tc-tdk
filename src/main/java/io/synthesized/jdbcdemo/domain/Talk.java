package io.synthesized.jdbcdemo.domain;

import lombok.Data;
import lombok.With;

import java.util.Set;

@Data
public class Talk {
    private final int id;
    private final String name;
    private final Conference conference;
    @With
    private final Status status;
    @With
    private final String feedback;
    private final Set<Speaker> speakers;
}
