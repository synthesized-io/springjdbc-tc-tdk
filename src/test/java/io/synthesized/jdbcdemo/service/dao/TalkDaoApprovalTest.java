package io.synthesized.jdbcdemo.service.dao;

import io.synthesized.jdbcdemo.domain.Conference;
import io.synthesized.jdbcdemo.domain.Talk;
import org.approvaltests.JsonApprovals;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.annotation.DirtiesContext;

import java.sql.SQLException;
import java.util.Objects;
import java.util.Set;

@SpringBootTest(classes = DaoTestConfiguration.class)
@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_CLASS)
public class TalkDaoApprovalTest {
    @Autowired
    private TalkDao dao;

    @Autowired
    ConferenceDao conferenceDao;

    private Conference conference;

    @BeforeEach
    void init() throws SQLException {
        conference = conferenceDao.getConferences().iterator().next();
    }

    @Test
    void getTalksByConference() {
        Objects.equals("a", "b");
        //Approval test
        Set<Talk> talks = dao.getTalksByConference(conference);
        JsonApprovals.verifyAsJson(talks);
    }

}
