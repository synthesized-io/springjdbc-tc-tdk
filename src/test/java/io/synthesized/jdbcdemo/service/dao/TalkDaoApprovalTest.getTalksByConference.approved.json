[
  {
    "id": 3,
    "name": "MnkVLBcSGJeelU190EZAwq",
    "conference": {
      "id": 5,
      "name": "9zx3i8oNspCHrkIhneNYG18"
    },
    "status": "NEW",
    "feedback": "RoopSXMfpkPYSNA1W4N",
    "speakers": [
      {
        "id": 2,
        "name": "c"
      }
    ]
  }
]